apps/trading/client-pages/markets/markets-page.tsx

```diff
@@ -125,78 +166,355 @@ export const MarketsPage = () => {
               </div>
             </Card>
           </div>
-          <MarketTables activeMarkets={activeMarkets} error={error} />
+          <MarketTables
+            markets={allMarkets}
+            marketAssets={marketAssets}
+            error={error}
+          />
         </div>
       </TinyScroll>
     </ErrorBoundary>
   );
 };
 
+type MarketType = 'FUTR' | 'PERP' | 'SPOT';
+type MarketState = 'OPEN' | 'PROPOSED' | 'CLOSED';
+
+type Filters = {
+  marketTypes: MarketType[];
+  marketStates: MarketState[];
+  assets: string[];
+  searchTerm: string | undefined;
+};
+type Actions = {
+  setMarketTypes: (marketTypes: MarketType[]) => void;
+  setMarketStates: (marketSates: MarketState[]) => void;
+  setAssets: (assets: string[]) => void;
+  setSearchTerm: (searchTerm: string) => void;
+  reset: () => void;
+};
+
+type MarketPageFiltersStore = Filters & Actions;
+
+const DEFAULT_FILTERS: Filters = {
+  marketTypes: [],
+  marketStates: ['OPEN'],
+  assets: [],
+  searchTerm: '',
+};
+
+const useMarketPageFilters = create<MarketPageFiltersStore>()((set) => ({
+  ...DEFAULT_FILTERS,
+  setMarketTypes: (marketTypes) => set({ marketTypes }),
+  setMarketStates: (marketStates) => set({ marketStates }),
+  setAssets: (assets) => set({ assets }),
+  setSearchTerm: (searchTerm) => set({ searchTerm }),
+  reset: () => set(DEFAULT_FILTERS),
+}));
+
+const filter = (
```

Comment:
non blocking: I was expecting to share the filter logic between market selector dropdown and the market list. Can we not have a single hook that manages this